Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -994,6 +994,29 @@ NiftiImageIO::SetImageIOMetadataFromNIfTI()
   std::ostringstream intent_name;
   intent_name << nim->intent_name;
   EncapsulateMetaData<std::string>(thisDic, "intent_name", intent_name.str());
+
+  // The below were added after ITK 5.3rc2.
+
+  EncapsulateMetaData<float>(thisDic, "qfac", nim->qfac);
+
+  std::vector<float> qto_xyz;
+  qto_xyz.push_back(nim->qto_xyz.m[0][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[0][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[0][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[0][3]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][3]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][3]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][3]);
```

Comment:
It appears that these 16 lines of code may be refactored to the following:

```
  for (const auto & row : nim->qto_xyz.m)
  {
    for (const float value : row)
    {
      qto_xyz.push_back(value);
    }
  }

```
Right?

Reply:
Or nested `for r`, `for c` loops.

Reply:
For loops are more readable and less prone to errors. Let the compiler do the unrolling, if advantageous.

Reply:
If speed is somehow the issue then we might make use of the `std::vector<>(begin, end)` constructor, as in
```cpp
std::vector<float> qto_xyz(&nim->qto_xyz.m[0][0], &nim->qto_xyz.m[0][0] + 16);
```
Admittedly this is ugly and does assume something about the packing of the `nim->qto_xyz.m` object.

Reply:
I think the nested r & c loops is best...

Reply:
I think it is OK to assume that `nim->qto_xyz.m` is a simple `float [4][4]` because that's just the way it is  😄 

Still I guess it would be preferable to store it as a 4x4 matrix, rather than an std::vector of 16 elements. Right?

Reply:
We could `static_assert` that `nim->qto_xyz.m` is of type `float [4][4]`.  But if speed is not critical here, the nested r & c loops do fine.

Reply:
Speed is not *at all* critical here. It's dwarfed by the file i/o to actually read the nifti file.

Reply:
Though not needed for the present case ... 
It might be nice, it might be nice
To have
```cpp
template <typename TUnderlying, int VRows, int VColumns>
constexpr inline std::vector<TUnderlying>
CArrayToStdVector(const TUnderlying (&arr)[VRows][VColumns])
{
  return { &arr[0][0], &arr[0][0] + VRows * VColumns };
}
```
on your side.

Reply:
Just made a pull request that would ease passing a C-style `float [4][4]` array directly to a `itk::Matrix`, please check: #3305  It would support the following:

    using MatrixType = Matrix<float, 4, 4>;
    EncapsulateMetaData<MatrixType>(thisDic, "qto_xyz", MatrixType(nim->qto_xyz.m));

Reply:
> ```c++
> template <typename TUnderlying, int VRows, int VColumns>
> constexpr inline std::vector<TUnderlying>
> CArrayToStdVector(const TUnderlying (&arr)[VRows][VColumns])
> {
>   return { &arr[0][0], &arr[0][0] + VRows * VColumns };
> }
> ```

@Leengit It might be nice indeed, but it's still not obvious to me why a fixed-size two-dimensional array (matrix) would need to be flattened to a 1-dimensional vector, before storing it into the dictionary. It seems more intuitive to me to have an `CArrayToStdArray(const TUnderlying (&arr)[VRows][VColumns])` function, returning an `std::array` of `std::array` elements.  (`std::array<std::array<TUnderlying, VColumns>, VRows>`). Right?

Reply:
> returning an `std::array` of `std::array` elements

That would be nice.